File: src/main/java/com/datastruct/arraystack/scala/Stack.scala
package com.datastruct.arraystack.scala

trait Stack[T] {
  // 获取栈大小
  def getSize: Int

  // 判断栈是否为空
  def isEmpty: Boolean

  // 返回栈顶元素（不取出）
  def top: T

  // 元素入栈
  def push(element: T): Unit

  // 元素出栈
  def pop: T
}

File: src/main/java/com/datastruct/arraystack/java/Stack.java
package com.datastruct.arraystack.java;

public interface Stack<T> {
    // 获取栈大小
    int getSize();

    // 栈是否为空
    boolean isEmpty();

    // 取出栈顶元素（但不删除）
    T top() throws ExceptionStackEmpty;

    // 进栈
    void push(T t) throws ExceptionArrayStackFull;

    // 出栈
    T pop() throws ExceptionStackEmpty;
}
